Mercury-Store/Mercury-Store/Home/Coordinator/HomeCoordinator.swift

```diff
@@ -7,59 +7,41 @@
 
 import UIKit
 
-class HomeCoordinator: HomeBaseCoordinator {
-    var parentCoordinator: MainBaseCoordinator?
+class HomeCoordinator : Coordinator {
     
-    lazy var rootViewController: UIViewController = UIViewController()
+    weak var parentCoordinator: Coordinator?
     
-    func start() -> UIViewController {
-        rootViewController = UINavigationController(rootViewController: HomeViewController(coordinator: self))
-        return rootViewController
-    }
+    var children: [Coordinator] = []
     
-    func moveTo(flow: AppFlow, userData: [String : Any]?) {
-        switch flow {
-        case .home(let homeScreen):
-            handleHomeFlow(for: homeScreen, userData: userData)
-        default:
-            parentCoordinator?.moveTo(flow: flow, userData: userData)
-        }
-    }
+    var navigationController: UINavigationController
     
-    private func handleHomeFlow(for screen: HomeScreen, userData: [String: Any]?) {
-        switch screen {
-        case .intialScreen:
-            navigationRootViewController?.popToRootViewController(animated: true)
-        case .searchScreen:
-            goToSearchScreenWith(title: "Search")
-        case .favouriteScreen:
-            //TODO: call goToFavouriteScreen()
-            break
-        case .productDetailScreen:
-            //TODO: call goToProductDetailScreenWtih(product: Product)
-            break
-        }
+    func start() {
+        print("HomeCoordinator Start")
+        
+        let homeVC = HomeViewController(nibName: String(describing: HomeViewController.self), bundle: nil)
         
+        homeVC.viewModel = HomeViewModel(homeFlow: self)
+        navigationController.pushViewController(homeVC, animated: true)
     }
     
-    func goToSearchScreenWith(title: String) {
-        resetToRoot()
-        navigationRootViewController?.pushViewController(SearchViewController(coordinator: self), animated: false)
+    init(navigationController : UINavigationController) {
+        self.navigationController = navigationController
     }
     
-    func goToFavouriteScreen() {
-        //TODO: navigate to favourite screen and push the view controller to the stack
+    deinit {
+        print("Deinit home coordinator")
     }
```

Comment:
```suggestion
```
No debug prints to be merged into main branches

Reply:
i was testing the life cycle of the coordinator I'm  gonna clear all the print statments